Sources/SwiftSyntaxMacros/SwiftSyntaxMacros.docc/SwiftSyntaxMacros.md

```diff
@@ -1,37 +1,308 @@
+      try SwitchExprSyntax("switch superset") {
+        for element in elements {
+          SwitchCaseSyntax(
+            """
+            case .\(element.identifier):
+              self = .\(element.identifier)
+            """
+          )
+        }
+        SwitchCaseSyntax("default: return nil")
+      }
+    }
+    
+    return [DeclSyntax(initializer)]
+  }
+}
+```
+
+## Inspect the SwiftSyntax Tree
+
+Navigating the syntax tree is not trivial. When the syntax tree seems overly complicated, this is usually because it needs to represent the entire Swift language. For example, each enum case declaration can declare multiple elements because it is possible to declare multiple enum elements after a single case keyword like:
+```swift
+case `actor`, `class`
+```
+
+Usually, the best way to explore the syntax trees is by example. <https://swift-ast-explorer.com> is a great webpage that allows the shows the syntax tree of entered Swift code.  
+
+> Experiment: Try pasting the following code to <https://swift-ast-explorer.com> and follow the syntax tree to understand the above code that extracts the name of the generic parameter in `EnumSubsetMacro.expansion`.
+> ```
+> @EnumSubset<Keyword>
+> ```
+
+Alternatively, the syntax tree can be printed in the debugger. When the debugger is stopped and `node` is a syntax node, the syntax tree can be printed using `po node`. This produces a syntax tree as shown in the [Macro Implementation](#Macro-Implementation) section.
+
+## Building Syntax Nodes
+
+There are three core approaches to build syntax nodes: Parsing from string literals, result builder initializers and memberwise initializers. 
+
+The `stringify` macro created its result type, an `ExprSyntax` from a string literal. As described in that section, this invokes the Swift parser to parse the contents of the string literal into a syntax tree. This technique is great for statically known trees or trees with a fixed number of parameters.
+
+The `EnumSubset` macro uses a result builder initializer to generate the `InitializerDeclSyntax` and `SwitchExprSyntax`. The initializer takes a header – the `swiftch` keyword and subject – and a trailing closure, which is a result builder that adds a `SwitchCaseSyntax` for every `case` item in the `switch` statement. Result builder initializers are a great tool for repetitive syntax constructs, like the `switch` in `EnumSubset`. Also note how `EnumSubsetMacro` combines result builders and string literals. It uses the result builder to generate the `SwitchExprSyntax` but the cases are constructed using string interpolation.
+
+Finally, every syntax type has a memberwise initializer with which a syntax node can be created by specifying all its children. This is the most low-level way of creating syntax nodes and has the most flexibility. It is also a good option for nodes that are not representable by string literals and don’t have result builder initializers.
```

Comment:
```suggestion
Finally, every syntax type has a memberwise initializer with which a syntax node can be created by specifying all its children.
```

Or reword the last two sentences - "good option" is a bit odd since it's actually the *only* option. Maybe:
> Finally, every syntax type has a memberwise initializer with which a syntax node can be created by specifying all its children. This initializer gives you full control when creating a syntax node, but is also the most verbose.